### PR TITLE
importccl: move to pkg/sql/importer

### PR DESCRIPTION
**Pull request by dt**

### CODE REVIEW

**Comment by dt** [comment]
Unfortunately I was like 85% done with this before I realized I still had my editor configured to use the more strict mvdan/gofumpt on every save rather than the usual gofmt / crlfmt, so there are some spurious / unrelated diffs in files I touched. They're all still crlfmt-compliant so they're not bad, and generally in fact I think it's rules are good which I why I usually use it, but it just adds some noise to a big change that I didn't intend. Getting them back out though it merge tedious (since, again, gofmt/crlfmt doesn't think they're wrong or change them) so unless there's an objection I was just going to go with it. The diffs are mostly things like trailing `}`  on its own line for multi-line struct literals or whatever.

**Comment by dt** [comment]
For now I haven't moved the `_` import to run the plan and job registration hooks from pkg/ccl to pkg/server yet just since at the moment that creates a cycle for the tests that are still in pkg/sql/importer but depend on pkg/server via testcluster. We'll need to do a pass where we split the testcluster-using tests out into importer_test before we can move the injection import to pkg/server, but I wanted to stop here before this PR got too much bigger.

**Comment by dt** [comment]
Okay, I think I fixed all the tests that weren't in `importer_test` to stop depending on testcluster and server directly and use the shim, so I was able to remove the ccl-side hook in favor of server now, so pure OSS binary should be able to run IMPORT as well.

**Comment by dt** [comment]
for the most part this is all mechanical code movement of whole files or whole functions as-is.

There were a couple though that required a little editing to work which might warrant closer review, namely:

- `move ingestsize setting to kv/bulk`: I removed some 'let's guess at the raft command overhead' that had no business being client-side. I almost got ride of the 'min-of-command-size-or-setting' logic entirely and may still say we should do so in a follow-up. You should just a) not change that setting b) keep it to values kv won't reject if you do, but we can do that in a targeted change.
- `move descriptor privilege setting to catalog/ingesting`: This required a little change to how we keep the temp system db perms handled one way and everything else another, changing how `wroteDbs` is populated.

**Comment by dt** [comment]
We could probably de-hook this planning entirely if we wanted at some point now but that seems like it can wait -- I mostly just wanted to get the code moved for now before the 22.1 branch is cut to minimize backport conflicts over the upcoming cycle.

**Review by adityamaru — APPROVED** [review]
Did one pass through all the commits, and it LGTM. Maybe we should get another stamp on this though considering the blast radius.

File: pkg/jobs/jobstest/traceutils.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Cockroach Authors.
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/jobs/jobstest/traceutils.go:1]: Ah, I thought the emitted events being checked here were trace events? but they're event log events, aren't they?

File: pkg/ccl/importccl/import_job.go [hunk 2]
```diff
@@ -564,9 +564,9 @@ func prepareNewTablesForIngestion(
 	// Write the new TableDescriptors and flip the namespace entries over to
 	// them. After this call, any queries on a table will be served by the newly
 	// imported data.
-	if err := backupccl.WriteDescriptors(ctx, p.ExecCfg().Codec, txn, p.User(), descsCol,
+	if err := ingesting.WriteDescriptors(ctx, p.ExecCfg().Codec, txn, p.User(), descsCol,
 		nil /* databases */, nil, /* schemas */
-		tableDescs, nil, tree.RequestedDescriptors, seqValKVs); err != nil {
+		tableDescs, nil, tree.RequestedDescriptors, seqValKVs, "" /* exemptParentName */); err != nil {
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/ccl/importccl/import_job.go:575]: these comments: still the worst.

File: pkg/jobs/jobstest/traceutils.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dt) [reply pkg/jobs/jobstest/traceutils.go]: renamed to logutils.go

File: pkg/ccl/importerccl/ccl_test.go [hunk 3]
```diff
@@ -0,0 +1,470 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/ccl/importerccl/ccl_test.go:7]: I _think_ the pattern is somewhat established, but added a comment.

**Comment by dt** [comment]
TFTR!

I'll wait until tomorrow to see if I get any other reviewers by then before I go OOO, but will probably aim to merge this one sooner rather than later with how conflict prone it is; while the files affected are numerous, I hope the actual blast radius is pretty small given it is mostly just movement and not changing behaviors. 

I'm sure we could get more reviewers to bikeshed some of the naming of the new packages, but I'd sorta rather do any of that in targeted, single-package follow-ups, once this one is merged with the actual untangling work it needed to do at once behind us.

**Comment by adityamaru** [comment]
SGTM, thanks!

**Comment by dt** [comment]
TFTR!

Checked with schema offline about the overall idea and what bits we were moving to catalog here and we’re broadly aligned on the bits that moved her moving and can refine more later.

let’s see if it can get by bors now.

bors r+

**Comment by adityamaru** [comment]
Could `single on` it to prevent it batching with other PRs.

**Comment by dt** [comment]
bors r+

bors single on

**Pull request merged**